public/js/analytics/typeBreakdown.js

```diff
@@ -0,0 +1,202 @@
+function GetColorCat(colors) {
+  if (colors.length === 0) {
+    return 'Colorless';
+  }
+  if (colors.length > 1) {
+    return 'Multi';
+  }
+  if (colors.length === 1) {
+    switch (colors[0]) {
+      case 'W':
+        return 'White';
+      case 'U':
+        return 'Blue';
+      case 'B':
+        return 'Black';
+      case 'R':
+        return 'Red';
+      case 'G':
+        return 'Green';
+      case 'C':
+      default:
+        return 'Colorless';
+    }
+  }
+}
+
+onmessage = (e) => {
+  if (!e) return;
+  const cards = e.data;
+
+  const TypeByColor = {
```

Comment:
```suggestion
  const TypeByColor = Object.fromEntries(['Creatures', /* ... */ ]).map((category) => ({ /* ... */ })))
```

Reply:
I was trying to think of a way to compress this down, thank you.